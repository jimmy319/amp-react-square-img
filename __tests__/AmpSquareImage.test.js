import React from 'react'
import renderer from 'react-test-renderer'
import AmpSquareImage, { statics } from '../dist/component'

describe('Test passing src to component', () => {
  test('[Normal Flow] Pass image url to component and DOM should be rendered with given src', () => {
    // arrangement
    const tree = renderer
      .create(<AmpSquareImage src='https://s.yimg.ir/q/test2.jpg' />)
      .toJSON()

    // action

    // assertion
    expect(tree).toMatchSnapshot()
  })
})

describe('Test resizingStrategy', () => {
  test('[Normal Flow] Pass FILL_RESIZING to component and `fill` class should be append onto DOM', () => {
    // arrangement
    const tree = renderer
      .create(<AmpSquareImage src='https://s.yimg.ir/q/test.jpg' resizingStrategy={statics.FILL_RESIZING} />)
      .toJSON()

    // action

    // assertion
    expect(tree).toMatchSnapshot()
  })

  test('[Normal Flow] Pass CONTAIN_RESIZING to component and `contain` class should be append onto DOM', () => {
    // arrangement
    const tree = renderer
      .create(<AmpSquareImage src='https://s.yimg.ir/q/test.jpg' resizingStrategy={statics.CONTAIN_RESIZING} />)
      .toJSON()

    // action

    // assertion
    expect(tree).toMatchSnapshot()
  })

  test('[Normal Flow] Pass COVER_RESIZING to component and `cover` class should be append onto DOM', () => {
    // arrangement
    const tree = renderer
      .create(<AmpSquareImage src='https://s.yimg.ir/q/test.jpg' resizingStrategy={statics.COVER_RESIZING} />)
      .toJSON()

    // action

    // assertion
    expect(tree).toMatchSnapshot()
  })

  test('[Normal Flow] Pass NONE_RESIZING to component and `none` class should be append onto DOM', () => {
    // arrangement
    const tree = renderer
      .create(<AmpSquareImage src='https://s.yimg.ir/q/test.jpg' resizingStrategy={statics.NONE_RESIZING} />)
      .toJSON()

    // action

    // assertion
    expect(tree).toMatchSnapshot()
  })

  test('[Normal Flow] Pass SCALE_DOWN_RESIZING to component and `scale-down` class should be append onto DOM', () => {
    // arrangement
    const tree = renderer
      .create(<AmpSquareImage src='https://s.yimg.ir/q/test.jpg' resizingStrategy={statics.SCALE_DOWN_RESIZING} />)
      .toJSON()

    // action

    // assertion
    expect(tree).toMatchSnapshot()
  })

  test('[Abnormal Flow] Pass UNKOWN_RESIZING to component and nothing being append onto DOM', () => {
    // arrangement
    const tree = renderer
      .create(<AmpSquareImage src='https://s.yimg.ir/q/test.jpg' resizingStrategy={statics.UNKOWN_RESIZING} />)
      .toJSON()

    // action

    // assertion
    expect(tree).toMatchSnapshot()
  })
})
