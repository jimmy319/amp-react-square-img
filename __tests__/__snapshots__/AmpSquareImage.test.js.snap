// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test passing src to component [Normal Flow] Pass image url to component and DOM should be rendered with given src 1`] = `
<div
  className="SquareImage__wrapper___3Vewc"
>
  <amp-img
    alt={undefined}
    attribution={undefined}
    class={undefined}
    layout="fill"
    sizes={undefined}
    src="https://s.yimg.ir/q/test2.jpg"
    srcset={undefined}
  />
</div>
`;

exports[`Test resizingStrategy [Abnormal Flow] Pass UNKOWN_RESIZING to component and nothing being append onto DOM 1`] = `
<div
  className="SquareImage__wrapper___3Vewc"
>
  <amp-img
    alt={undefined}
    attribution={undefined}
    class={undefined}
    layout="fill"
    sizes={undefined}
    src="https://s.yimg.ir/q/test.jpg"
    srcset={undefined}
  />
</div>
`;

exports[`Test resizingStrategy [Normal Flow] Pass CONTAIN_RESIZING to component and \`contain\` class should be append onto DOM 1`] = `
<div
  className="SquareImage__wrapper___3Vewc"
>
  <amp-img
    alt={undefined}
    attribution={undefined}
    class="SquareImage__contain___3Nskz"
    layout="fill"
    sizes={undefined}
    src="https://s.yimg.ir/q/test.jpg"
    srcset={undefined}
  />
</div>
`;

exports[`Test resizingStrategy [Normal Flow] Pass COVER_RESIZING to component and \`cover\` class should be append onto DOM 1`] = `
<div
  className="SquareImage__wrapper___3Vewc"
>
  <amp-img
    alt={undefined}
    attribution={undefined}
    class="SquareImage__cover___lDXKz"
    layout="fill"
    sizes={undefined}
    src="https://s.yimg.ir/q/test.jpg"
    srcset={undefined}
  />
</div>
`;

exports[`Test resizingStrategy [Normal Flow] Pass FILL_RESIZING to component and \`fill\` class should be append onto DOM 1`] = `
<div
  className="SquareImage__wrapper___3Vewc"
>
  <amp-img
    alt={undefined}
    attribution={undefined}
    class="SquareImage__fill___bdZjK"
    layout="fill"
    sizes={undefined}
    src="https://s.yimg.ir/q/test.jpg"
    srcset={undefined}
  />
</div>
`;

exports[`Test resizingStrategy [Normal Flow] Pass NONE_RESIZING to component and \`none\` class should be append onto DOM 1`] = `
<div
  className="SquareImage__wrapper___3Vewc"
>
  <amp-img
    alt={undefined}
    attribution={undefined}
    class="SquareImage__none___jq2dA"
    layout="fill"
    sizes={undefined}
    src="https://s.yimg.ir/q/test.jpg"
    srcset={undefined}
  />
</div>
`;

exports[`Test resizingStrategy [Normal Flow] Pass SCALE_DOWN_RESIZING to component and \`scale-down\` class should be append onto DOM 1`] = `
<div
  className="SquareImage__wrapper___3Vewc"
>
  <amp-img
    alt={undefined}
    attribution={undefined}
    class="SquareImage__scale-down___2YrR-"
    layout="fill"
    sizes={undefined}
    src="https://s.yimg.ir/q/test.jpg"
    srcset={undefined}
  />
</div>
`;
